screwdriver/screwdriver.go

```diff
@@ -0,0 +1,196 @@
+package screwdriver
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+)
+
+const (
+	apiVersion        = "v4"
+	validatorEndpoint = "validator"
+	tokenEndpoint     = "auth/token"
+)
+
+// API has method to get job
+type API interface {
+	Job(jobName, filePath string) (Job, error)
+	JWT() string
+}
+
+type sdAPI struct {
+	HTTPClient *http.Client
+	UserToken  string
+	APIURL     string
+	SDJWT      string
+}
+
+var _ API = (*sdAPI)(nil)
+
+// Step is step entity struct
+type Step struct {
+	Name    string `json:"name"`
+	Command string `json:"command"`
+}
+
+// Job is job entity struct
+type Job struct {
+	Steps       []Step            `json:"commands"`
+	Environment map[string]string `json:"environment"`
+	Image       string            `json:"image"`
+}
+
+type jobs map[string][]Job
+
+type validatorResponse struct {
+	Jobs   jobs     `json:"jobs"`
+	Errors []string `json:"errors"`
+}
+
+type tokenResponse struct {
+	JWT string `json:"token"`
+}
+
+// New creates a API
+func New(apiURL, token string) (API, error) {
+	s := &sdAPI{
+		HTTPClient: http.DefaultClient,
+		APIURL:     apiURL,
+		UserToken:  token,
+	}
+
+	jwt, err := s.jwt()
+	if err != nil {
+		return s, err
+	}
+
+	s.SDJWT = jwt
+
+	return s, nil
+}
+
+func (sd *sdAPI) makeURL(endpoint string) (*url.URL, error) {
+	u, err := url.Parse(sd.APIURL)
```

Comment:
Is it better to call `url.Parse()` in `New()` function?
Both `jwt` and `validate` requires `url.Parse(sd.APIURL)`, but the results are always same.

Reply:
Like [this snippet](https://play.golang.org/p/MvbR9JR0K_U), `path.Join` replace `//` in schema(`https://`)to `/`.
To avoid this case, We need to use `url.Parse` and join `Url.Path` and endpoint.

If it is called in `New` function, I think both `jwt` and `validate` should clone vanilla `Url` parsed from `sd.APIURL`. I think it is so verbose.

Reply:
I see.